File: apple/Sources/Shared/Views/ChatView.swift
import SwiftUI

/// Minimal grounded-chat surface. The backend handles input screening, crisis
/// detection, PII redaction, and BYOK routing — the client only sends the
/// turn history and renders the normalized text reply.
struct ChatView: View {
    @State private var messages: [ChatMessage] = []
    @State private var input = ""
    @State private var sending = false
    @State private var error: String?

    var body: some View {
        NavigationStack {
            VStack(spacing: 0) {
                ScrollViewReader { proxy in
                    ScrollView {
                        LazyVStack(spacing: 10) {
                            if messages.isEmpty {
                                Text("chat.empty").font(.callout)
                                    .foregroundStyle(Theme.textMuted)
                                    .padding(.top, 40)
                            }
                            ForEach(messages) { msg in
                                bubble(msg).id(msg.id)
                            }
                            if sending { typingIndicator }
                        }
                        .padding(16)
                        .frame(maxWidth: 700)
                        .frame(maxWidth: .infinity)
                    }
                    .onChange(of: messages.count) { _ in
                        if let last = messages.last { withAnimation { proxy.scrollTo(last.id, anchor: .bottom) } }
                    }
                }

                if let error {
                    Text(error).font(.caption).foregroundStyle(Theme.red)
                        .padding(.horizontal, 16).padding(.bottom, 4)
                }

                composer
            }
            .background(Theme.bg)
            .navigationTitle("tab.chat")
        }
    }

    private func bubble(_ msg: ChatMessage) -> some View {
        let isUser = msg.role == "user"
        return HStack {
            if isUser { Spacer(minLength: 40) }
            Text(msg.content)
                .font(.callout)
                .foregroundStyle(isUser ? Color.white : Theme.textPrimary)
                .padding(.horizontal, 12).padding(.vertical, 8)
                .background(isUser ? Theme.blue.opacity(0.8) : Theme.card)
                .overlay(RoundedRectangle(cornerRadius: 12).stroke(Theme.cardBorder, lineWidth: isUser ? 0 : 1))
                .clipShape(RoundedRectangle(cornerRadius: 12))
                .textSelection(.enabled)
            if !isUser { Spacer(minLength: 40) }
        }
    }

    private var typingIndicator: some View {
        HStack { ProgressView().controlSize(.small); Spacer() }
    }

    private var composer: some View {
        HStack(spacing: 8) {
            TextField("chat.placeholder", text: $input, axis: .vertical)
                .textFieldStyle(.roundedBorder)
                .lineLimit(1...5)
                .onSubmit(send)
            Button(action: send) {
                Image(systemName: "arrow.up.circle.fill").font(.title2)
            }
            .disabled(input.trimmingCharacters(in: .whitespaces).isEmpty || sending)
        }
        .padding(12)
        .background(Theme.bg)
    }

    private func send() {
        let text = input.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !text.isEmpty, !sending else { return }
        input = ""
        error = nil
        messages.append(ChatMessage(role: "user", content: text))
        sending = true
        Task {
            defer { sending = false }
            do {
                let resp = try await APIClient.shared.chat(messages: messages, tier: "medium")
                if let err = resp.error?.message {
                    error = err
                } else {
                    let reply = resp.firstText
                    messages.append(ChatMessage(role: "assistant", content: reply.isEmpty ? String(localized: "chat.no_reply") : reply))
                }
            } catch {
                self.error = (error as? APIError)?.errorDescription ?? error.localizedDescription
            }
        }
    }
}

File: windows/CollegeCounselor/ChatView.cs
using System.Windows;
using System.Windows.Controls;
using System.Windows.Input;
using System.Windows.Media;

namespace CollegeCounselor;

/// Minimal grounded chat. The backend does input screening, crisis detection,
/// PII redaction, and BYOK routing; the client sends turns and renders replies.
internal sealed class ChatView : UserControl
{
    private readonly StackPanel _messages = new() { Margin = new Thickness(16) };
    private readonly ScrollViewer _scroll;
    private readonly TextBox _input = Theme.Input();
    private readonly Button _send = Theme.Button("Send", Theme.Blue);
    private readonly List<(string role, string content)> _history = new();

    public ChatView()
    {
        _scroll = new ScrollViewer { Content = _messages, VerticalScrollBarVisibility = ScrollBarVisibility.Auto, Background = Theme.Bg };
        _messages.Children.Add(Theme.Label("Ask about your application, essays, schools, or deadlines.", 13, Theme.TextMuted));

        _input.Margin = new Thickness(0);
        _input.AcceptsReturn = false;
        _input.KeyDown += (_, e) => { if (e.Key == Key.Enter) { e.Handled = true; OnSend(this, new RoutedEventArgs()); } };
        _send.Margin = new Thickness(8, 0, 0, 0);
        _send.Click += OnSend;

        var inputRow = new DockPanel { Margin = new Thickness(12), LastChildFill = true };
        DockPanel.SetDock(_send, Dock.Right);
        inputRow.Children.Add(_send);
        inputRow.Children.Add(_input);

        var dock = new DockPanel { LastChildFill = true, Background = Theme.Bg };
        DockPanel.SetDock(inputRow, Dock.Bottom);
        dock.Children.Add(inputRow);
        dock.Children.Add(_scroll);
        Content = dock;
    }

    private async void OnSend(object sender, RoutedEventArgs e)
    {
        var text = _input.Text.Trim();
        if (text.Length == 0) return;
        _input.Clear();
        AddBubble(text, isUser: true);
        _history.Add(("user", text));
        _send.IsEnabled = false;
        try
        {
            var recent = _history.Count > 30 ? _history.GetRange(_history.Count - 30, 30) : _history;
            var resp = await ApiClient.Shared.ChatAsync(recent, "medium");
            var reply = resp.FirstText;
            if (string.IsNullOrWhiteSpace(reply)) reply = "(No response)";
            _history.Add(("assistant", reply));
            AddBubble(reply, isUser: false);
        }
        catch (Exception ex)
        {
            AddBubble(ex.Message, isUser: false, error: true);
        }
        finally { _send.IsEnabled = true; }
    }

    private void AddBubble(string text, bool isUser, bool error = false)
    {
        var bubble = new Border
        {
            Background = isUser ? Tint(Theme.Blue, 0.85) : Theme.Card,
            BorderBrush = Theme.CardBorder,
            BorderThickness = new Thickness(isUser ? 0 : 1),
            CornerRadius = new CornerRadius(12),
            Padding = new Thickness(12, 8, 12, 8),
            Margin = new Thickness(0, 5, 0, 5),
            MaxWidth = 460,
            HorizontalAlignment = isUser ? HorizontalAlignment.Right : HorizontalAlignment.Left,
            Child = new TextBlock
            {
                Text = text,
                TextWrapping = TextWrapping.Wrap,
                Foreground = error ? Theme.Red : (isUser ? Brushes.White : Theme.TextPrimary),
                FontSize = 13,
            },
        };
        _messages.Children.Add(bubble);
        _scroll.ScrollToEnd();
    }

    private static SolidColorBrush Tint(Brush b, double alpha)
    {
        var c = ((SolidColorBrush)b).Color;
        return new SolidColorBrush(Color.FromArgb((byte)(alpha * 255), c.R, c.G, c.B));
    }
}
